src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepository.cs

```diff
@@ -882,17 +882,21 @@ private bool ParallelProcessBlock(ProcessBlocksInfo round, Block block, ChainedH
 
                                 conn.Rollback();
 
+                                throw;
```

Comment:
These changes appears to be unnecessary. Previously, we would end batch in the catch (which catches all exceptions) so moving it to the `finally` doesn't really add any value? Unless it is moved there to ensure that it always calls end batch in non-error scenarios? Or is that what the extra `else` is for further down?

Unless I'm missing something here 😎

Reply:
The idea is to always call `EndBatch` where errors in `Rollback` could previously prevent that.

Reply:
Ah ok yes, cool :)